src/support/generation.c

```diff
@@ -80,8 +78,7 @@ uint64_t
  *	Wait for the resource to drain.
  */
 void
-	TABBED IN to avoid dist/ functions:
-	__wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
+__wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
 {
```

Comment:
We could, and I agree with the goal.

I'm hesitant to make this change: if the cache isn't effective (which, if there's a lot of churn, I wouldn't expect it to be?), this change means we'd repeatedly walk the sessions until we succeed, rather than walking them once and waiting at each slot until we're past it. I don't have a sense if that's likely or a problem, you've done more split-generation stuff than I have. Let's talk this over, happy to make this change if you want.

Is your thought to cache the oldest session in the connection, and use atomic instructions to update it after a walk?

(I'm still planning to go back and finish off the read/write lock change, fwiw.)